### PR TITLE
Gw collector

### PR DESCRIPTION
**Pull request by ajssmith**
add role for gateways and introduce simple aging of completed flow records (default 30 minutes)

### CODE REVIEW

File: api/types/client.go [hunk 1]
```diff
@@ -93,8 +93,9 @@ type SiteConfigSpec struct {
 	SiteTtl             time.Duration
 	EnableConsole       bool
 	EnableFlowCollector bool
+	FlowRecordTtl       time.Duration
```
Comment (grs) [inline api/types/client.go:96]: Could put this under FlowControllerOptions, but that is very minor.

**Review by grs — APPROVED** [review]

**Review by nluaces — APPROVED** [review]

File: cmd/skupper/skupper_kube_site.go [hunk 2]
```diff
@@ -125,6 +128,7 @@ func (s *SkupperKubeSite) CreateFlags(cmd *cobra.Command) {
 	cmd.Flags().BoolVarP(&routerCreateOpts.EnableServiceSync, "enable-service-sync", "", true, "Participate in cross-site service synchronization")
 	cmd.Flags().DurationVar(&routerCreateOpts.SiteTtl, "service-sync-site-ttl", 0, "Time after which stale services, i.e. those whose site has not been heard from, created through service-sync are removed.")
 	cmd.Flags().BoolVarP(&routerCreateOpts.EnableFlowCollector, "enable-flow-collector", "", false, "Enable cross-site flow collection for the application network")
+	cmd.Flags().DurationVar(&routerCreateOpts.FlowRecordTtl, "flow-collector-record-ttl", 0, "Time after which terminated flow records are deleted, i.e. those flow records that have an end time set.")
```
Comment (fgiorgetti) [inline cmd/skupper/skupper_kube_site.go:131]: One minor suggestion to either set the default value to 30m or say what the default is in the help message.

**Review by fgiorgetti — APPROVED** [review]

File: api/types/client.go (hunk 1 above)
Comment (ajssmith) [reply api/types/client.go]: agree, will update

File: cmd/skupper/skupper_kube_site.go (hunk 2 above)
Comment (ajssmith) [reply cmd/skupper/skupper_kube_site.go]: good suggestion, updated

**Pull request merged**